app/assets/javascripts/discourse/app/controllers/edit-user-directory-columns.js

```diff
@@ -0,0 +1,98 @@
+import Controller from "@ember/controller";
+import ModalFunctionality from "discourse/mixins/modal-functionality";
+import { ajax } from "discourse/lib/ajax";
+import EmberObject, { action } from "@ember/object";
+import { extractError } from "discourse/lib/ajax-error";
+import { reload } from "discourse/helpers/page-reloader";
+
+const UP = "up";
+const DOWN = "down";
+
+export default Controller.extend(ModalFunctionality, {
+  loading: true,
+  columns: null,
+  labelKey: null,
+
+  onShow() {
+    ajax("directory-columns.json")
+      .then((response) => {
+        this.setProperties({
+          loading: false,
+          columns: response.directory_columns
+            .sort((a, b) => (a.position > b.position ? 1 : -1))
+            .map((c) => EmberObject.create(c)),
+        });
+      })
+      .catch(extractError);
+  },
+
+  @action
+  save() {
+    this.set("loading", true);
+    const data = {
+      directory_columns: this.columns.map((c) =>
+        c.getProperties("id", "enabled", "position")
+      ),
+    };
+
+    ajax("directory-columns.json", { type: "PUT", data })
+      .then(() => {
+        reload();
+      })
+      .catch(extractError);
+  },
+
+  @action
+  resetToDefault() {
+    let resetColumns = this.columns;
+    resetColumns
+      .sort((a, b) =>
+        (a.automatic_position || a.user_field.position + 1000) >
```

Comment:
This sorts all the columns with `automatic_position` to the top, then orders the user_field columns properly based on their position. If we ever got over 1000 automatic columns by default we have lost our way and are out of business